direct/core/src/main/java/cz/o2/proxima/direct/commitlog/OffsetExternalizer.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2017-2021 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.direct.commitlog;
+
+import cz.o2.proxima.scheme.SerializationException;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.List;
+
+/** Interface provides methods for serializing offsets to external formats. */
+public interface OffsetExternalizer<OffsetT extends Offset> {
+
+  /**
+   * Serializes offsets to bytes, default implementation uses native java serialization.
+   *
+   * @param offsets list to serialize.
+   * @return serialized offsets in bytes
+   */
+  default byte[] toBytes(List<OffsetT> offsets) {
```

Comment:
The first implementation was without List, but  I have changed it because all methods of ObserveHandle uses List<Offset>. No problem to revert it back to single Offset.